trpc/router/goalV2.ts

```diff
@@ -160,8 +160,17 @@ export const goal = router({
                 offset,
                 ...ctx.session.user,
             });
+            const { starred, watching } = { ...goalsQuery };
 
-            const goals = await query.$castTo<Goal>().execute();
+            let goals;
+
+            if (starred) {
+                goals = await query.$castTo<Goal>().where('_isStarred', '=', true).execute();
```

Comment:
Again bad decision, here you again get most shorter list of goals

Needs add filter by `_goalStargizers` and `_goalWatchers` which contains link between goal entity by `id` and user by `activityId`

These filter needs add in `getAllGoalsQuery`

Reply:
like ex.
```js
     ...
     .$if(goalsQuery.starred, (qb) => qb.where('Goal.id', 'in', ({ selectFrom }) =>
         selectFrom('_goalStargizers').select('B').where('A', '=', activityId),
      ))
```